newspaper-api-gateway/app/controllers/ObservationController.scala

```diff
@@ -0,0 +1,43 @@
+package controllers
+
+import java.net.{MalformedURLException, URL}
+import javax.inject.Inject
+
+import model.PageToObserve
+import model.Subscriber._
+import play.api.libs.json.{JsSuccess, JsValue}
+import play.api.mvc._
+import services.OutboundService
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.util.Try
+
+class ObservationController @Inject()(
+       out: OutboundService,
+       cc: ControllerComponents) extends AbstractController(cc) {
+
+  def add() = Action.async(parse.json) { implicit request: Request[JsValue] =>
+    request.body.validate[PageToObserve] match {
+      case JsSuccess(obs, _) if checkUrlValidness(obs.pageUrl).getOrElse(false) =>
+        val prefixedObserver = if(obs.pageUrl.contains("http"))
```

Comment:
You already checked URL's validity by instantiating a `java.net.URL` - maybe you could use it here instead of doing manual string operations?